### PR TITLE
Don't require elasticsearch for Kibana resources

### PR DESCRIPTION
**Pull request by tobio**
Fixes https://github.com/elastic/terraform-provider-elasticstack/issues/345

Kibana spaces shouldn't need Elasticsearch to be configured to work.

### CODE REVIEW

**Review by dimuon — APPROVED** [review]
LGTM 👍, it just needs changelog :)

**Comment by dimuon** [comment]
Still, it looks like we also have to change [the way](https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/clients/api_client.go#L704) kibana client is built.

File: internal/kibana/space.go [hunk 1]
```diff
@@ -136,11 +132,14 @@ func resourceSpaceRead(ctx context.Context, d *schema.ResourceData, meta interfa
 	if diags.HasError() {
 		return diags
 	}
-	compId, diags := clients.CompositeIdFromStr(d.Id())
-	if diags.HasError() {
-		return diags
+	id := d.Id()
+	if strings.Contains(id, "/") {
```
Comment (dimuon) [inline internal/kibana/space.go]: It's a trivial check, but maybe it makes sense to create a helper function (e.g. `IsCompositeId`) next to `func CompositeIdFromStr` to keep the details of composite ID creation in one place.

**Comment by tobio** [comment]
@dimuon I might defer this one until https://github.com/elastic/terraform-provider-elasticstack/pull/343 is done since it will simplify the creation changes.

**Review by dimuon — APPROVED** [review]
LGTM

**Pull request merged**